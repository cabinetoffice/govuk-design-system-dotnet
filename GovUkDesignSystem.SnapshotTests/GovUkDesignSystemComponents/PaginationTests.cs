using System.Collections.Generic;
using System.Threading.Tasks;
using GovUkDesignSystem.GovUkDesignSystemComponents;
using GovUkDesignSystem.SnapshotTests.Helpers;
using Xunit;

namespace GovUkDesignSystem.SnapshotTests.GovUkDesignSystemComponents
{
    public class PaginationTests : SnapshotTestBase
    {
        private PaginationViewModel DefaultPaginationViewModel()
        {
            return new PaginationViewModel
            {
                Previous = new PaginationPreviousViewModel
                {
                    Text = "Prev text",
                    LabelText = "Prev label text",
                    Href = "#"
                }, 
                Next = new PaginationNextViewModel 
                { 
                    Text = "Next text",
                    LabelText = "Next label text",
                    Href = "#"
                },
                Classes = "pagination-class",
                LandmarkLabel = "landmarkLabel",
                Items = new List<PaginationItemViewModel>
                {
                    new PaginationItemViewModel
                    {
                        Number = "1",
                        VisuallyHiddenText = "1",
                        Href = "#",
                    },
                    new PaginationItemViewModel
                    {
                        Ellipsis = true
                    },
                    new PaginationItemViewModel
                    {
                        Number = "3",
                        VisuallyHiddenText = "3",
                        Href = "#",
                        Current = true
                    },
                    new PaginationItemViewModel
                    {
                        Number = "4",
                        VisuallyHiddenText = "4",
                        Href = "#"
                    }
                }
            };
        }
        
        [Fact]
        public async Task Render_AllValues()
        {
            // Act & Assert
            await VerifyPartial("Pagination", DefaultPaginationViewModel());
        }
        
        [Fact]
        public async Task Render_NoItems()
        {
            // Arrange
            var viewModel = DefaultPaginationViewModel();
            viewModel.Items = null;
            
            // Act & Assert
            await VerifyPartial("Pagination", viewModel);
        }
        
        [Fact]
        public async Task Render_LastPage()
        {
            // Arrange
            var viewModel = DefaultPaginationViewModel();
            viewModel.Items.RemoveAt(3);
            viewModel.Next = null;
            
            // Act & Assert
            await VerifyPartial("Pagination", viewModel);
        }
        
        [Fact]
        public async Task Render_FirstPage()
        {
            // Arrange
            var viewModel = DefaultPaginationViewModel();
            viewModel.Items[2].Current = false;
            viewModel.Items[0].Current = true;
            viewModel.Previous = null; 
            
            // Act & Assert
            await VerifyPartial("Pagination", viewModel);
        }
    }
}