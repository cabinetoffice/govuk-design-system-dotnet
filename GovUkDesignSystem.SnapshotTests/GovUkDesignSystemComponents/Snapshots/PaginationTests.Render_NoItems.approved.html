<nav class="govuk-pagination pagination-class govuk-pagination--block" role="navigation" aria-label="landmarkLabel">
  <div class="govuk-pagination__prev">
    <a class="govuk-link govuk-pagination__link" href="#" rel="prev">
      <svg class="govuk-pagination__icon govuk-pagination__icon--prev" xmlns="http://www.w3.org/2000/svg" height="13" width="15" aria-hidden="true" focusable="false" viewBox="0 0 15 13">
        <path d="m6.5938-0.0078125-6.7266 6.7266 6.7441 6.4062 1.377-1.449-4.1856-3.9768h12.896v-2h-12.984l4.2931-4.293-1.414-1.414z"></path>
      </svg>
      <span class="govuk-pagination__link-title ">
                    Prev text
                </span>
      <span class="govuk-visually-hidden">:</span>
      <span class="govuk-pagination__link-label">Prev label text</span>
    </a>
  </div>
  <div class="govuk-pagination__next">
    <a class="govuk-link govuk-pagination__link" href="#" rel="next">
      <svg class="govuk-pagination__icon govuk-pagination__icon--next" xmlns="http://www.w3.org/2000/svg" height="13" width="15" aria-hidden="true" focusable="false" viewBox="0 0 15 13">
        <path d="m8.107-0.0078125-1.4136 1.414 4.2926 4.293h-12.986v2h12.896l-4.1855 3.9766 1.377 1.4492 6.7441-6.4062-6.7246-6.7266z"></path>
      </svg>
      <span class="govuk-pagination__link-title ">
                    Next text
                </span>
      <span class="govuk-visually-hidden">:</span>
      <span class="govuk-pagination__link-label">Next label text</span>
    </a>
  </div>
</nav>